### ISSUE TITLE
Use `data_train` in validation curve

### ISSUE THREAD

**Issue by ArturoAmorQ** [INRIA/scikit-learn-mooc#809]
As reported in [this forum post](https://mooc-forums.inria.fr/moocsl/t/use-of-data-or-data-train/15880), in the instructions for the validation curve in the [Exercise M6.03](https://inria.github.io/scikit-learn-mooc/python_scripts/ensemble_sol_03.html) it says 
>For both the gradient-boosting and random forest models, create a validation curve using the training set to assess the impact of the number of trees on the performance of each model.

But then in the solution we don't use the `data_train` but all data for the validation curve.

Related to #603, we can take the opportunity to also emphasize at this stage that `n_estimators` should not be tuned for random forests, and that the validation curve in the notebook was only computed for educational purposes.

**Comment by fritshermans** [comment]
@ArturoAmorQ, as the HistGradientBoostingClassifier/Regressor by default applies only auto-stopping when there are at least 10k samples, it would make sense to do explain this behavior?

**Comment by ArturoAmorQ** [comment]
You are right, that would make sense. Maybe we can additionally point students towards this piece of documentation: [Effect of number of trees and early stopping](https://scikit-learn.org/stable/auto_examples/ensemble/plot_hgbt_regression.html#effect-of-number-of-trees-and-early-stopping).

**Comment by fritshermans** [comment]
Apologies @ArturoAmorQ, I realize now that a `GradientBoostingClassifier` is used, not the histogram based version of it. Let me add the validation curve for GBT as that is what the exercise prescibes.

**Comment by fritshermans** [comment]
I think this issue could be closed by [PR 827](https://github.com/INRIA/scikit-learn-mooc/pull/827)

**Comment by ArturoAmorQ** [comment]
You are right. Thanks for pointing this out.
